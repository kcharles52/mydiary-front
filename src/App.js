import React, { Component,Fragment } from "react";
import AppRoutes from "./routes";
import { Provider } from "react-redux";
import store from "./store/";

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Fragment>
          <AppRoutes />
        </Fragment>
      </Provider>
    );
  }
}

export default App;
