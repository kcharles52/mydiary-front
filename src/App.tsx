
//react libraries
import * as React from "react";

//third party packages
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.min.css";


//components
import AppRoutes from "./routes";

const App =()=>(
  <React.Fragment>
    <ToastContainer />
    <AppRoutes />
  </React.Fragment>
);
export default App;
