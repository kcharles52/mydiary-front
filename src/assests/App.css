.App {
  text-align: center;
}

/*General settings*/ 
body {
  background: #fff;
  font-family: "Open Sans", sans-serif;
  color: #4e4e4e;
  line-height: 22px;
  margin: 0;
}

*,
*:before,
*:after {
  box-sizing: border-box;
}

.bg {
  background-color: #a75110;
  color: #fff !important;
}

h1,
h2 {
  font-weight: 600;
  font-family: "Open Sans", sans-serif;
  color: #4e4e4e;
}

h1 {
  font-size: 36px;
  color: #fff;
}

h2 {
  font-size: 20px;
}

h3 {
  font-size: 16px;
  color: #787878;
  font-weight: 400;
  line-height: 24px;
}

a {
  color: #c46731;
  text-decoration: none;
  padding: 0;
  margin: 0;
}

a:hover,
a:focus {
  color: #ffffff;
  text-decoration: none;
}

.button,
#button {
  border: 0;
  outline: none;
  border-radius: 0;
  padding: 15px 5px;
  font-size: 2rem;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
  letter-spacing: 0.1em;
  background: #a75114;
  color: #ffffff;
}

.button:hover,
.button:focus,
#button:hover,
#button:focus {
  background: #db8a5c;
}

.button-block {
  display: block;
  width: 100%;
}

.forgot {
  margin-top: -20px;
  text-align: right;
}

.center h2 {
  font-size: 30px;
  margin-top: 0px;
  margin-bottom: 20px;
}

body>section {
  padding: 70px 0;
}

section {
  min-height: 500px;
  padding: 0 70px;
}

.center {
  text-align: center;
  /*! padding-bottom: 50px; */
}

/*Header styling*/
.navbar {
  background-color: rgb(49, 31, 24, 0.8);
  padding: 15px 0px 10px 0px;
  overflow: auto;
}

.navbar h1 {
  font-size: 40px;
  float: left;
  padding: 0 70px;
  margin-bottom: 0;
}

.navbar>nav {
  float: right;
  margin-top: -18px;
}

.navbar>nav li {
  margin-left: 35px;
  padding-bottom: 18px;
  float: left;
  list-style: none;
  position: relative;
  clear: right;
}

.navbar nav li>a {
  padding: 5px 12px;
  margin: 0;
  border-radius: 3px;
  color: #fff;
  line-height: 24px;
  display: in-block;
}

.navbar nav li>a:hover {
  background-color: #c46731;
  color: #fff;
}

.navbar nav li>.active>a,
.navbar nav li>.active>a:hover,
.navbar nav li>.active>a:focus {
  background-color: #c46731;
  color: #fff;
}


/*Form styling*/

.form {
  background: rgb(37, 27, 26, 0.8);
  padding: 40px;
  max-width: 500px;
  border-radius: 4px;
  margin-left: 100px;
}

.tab-group {
  list-style: none;
  padding: 0;
  margin: 0 0 40px 0;
}

.tab-group:after {
  content: "";
  display: table;
  clear: both;
}

.tab-group div {
  display: inline;
}

.tab-group div a {
  display: block;
  text-decoration: none;
  padding: 15px;
  background: rgba(160, 179, 176, 0.25);
  color: #a0b3b0;
  font-size: 20px;
  float: left;
  width: 50%;
  text-align: center;
  cursor: pointer;
}

.tab-group div a:hover {
  background: #db8a5c;
  color: #ffffff;
}

.tab-group .active a {
  background: #a75114;
  color: #ffffff;
}

.tab-content>div:last-child {
  display: none;
}

h1,
h2 {
  text-align: center;
  color: #e9d18f;
  font-weight: 300;
  margin: 0 0 40px;
}

label {
  top: -22px;
  color: #e9d18f;
}

label .req {
  margin: 2px;
  color: #db8a5c;
}

label.active {
  left: 2px;
  font-size: 14px;
}

label.active .req {
  opacity: 0;
}

label.highlight {
  color: #ffffff;
}

input,
textarea {
  font-size: 22px;
  display: block;
  width: 100%;
  height: 100%;
  padding: 5px 10px;
  background: none;
  background-image: none;
  border: 1px solid #db8a5c;
  /* color: #ffffff; */
  border-radius: 0;
}

input:focus,
textarea:focus {
  outline: 0;
  border-color: #b7a170;
}

textarea {
  border: 2px solid #db8a5c;
  resize: vertical;
}

.field-wrap {
  position: relative;
  margin-bottom: 40px;
}

.top-row:after {
  content: "";
  display: table;
  clear: both;
}

.top-row>div {
  float: left;
  width: 48%;
  margin-right: 4%;
}

.top-row>div:last-child {
  margin: 0;
}

/*End of form styling*/

/*styling for the footer*/
#footer {
  background: #e9d180;
  border-top: 5px solid #508f10;
  font-size: 14px;
}

#footer .center {
  padding: 20px 0;
}


/* Style entries page */

.tab {
  overflow: hidden;
  color: #fff;
  /*! border: 1px solid #ccc; */
  background-color: #853700;
}

.tab button {
  background-color: inherit;
  float: left;
  border: none;
  outline: none;
  cursor: pointer;
  padding: 14px 16px;
  transition: 0.3s;
  color: #fff;
}

.tab button:hover {
  background-color: #ddd;
}

.tab button.active {
  background-color: #ccc;
}

/* Style the tab content */

.tabcontent {
  display: none;
  padding: 6px 12px;
  border: 1px solid #ccc;
  border-top: none;
}

/*diary entries page*/

/* .d-table {
   padding: 20px;
   background-image: url("../img/dEntryBg.jpg");
   background-repeat: no-repeat;
} */

.d-table table {
  padding: 2%;
  background-color: rgb(50, 50, 50, 0.9);
  color: white;
}

table {
  border-collapse: collapse;
}

td,
th {
  border: 1px solid #fff;
  padding: 15px 20px;
}

.action a {
  padding: 5px 10px;
  border: 2px solid;
  margin-right: 3px;
}

.entryCards {
  margin: auto;
  width: 90%;
  overflow: auto;
  height: 80vh;
  /* padding: 0 30px; */
}

.dEntryH {
  max-width: 33%;
  width: 30%;
  height: 300px;
  background-color: rgb(20, 20, 30, 0.9);
  padding: 3%;
  margin: 10px;
  position: relative;
  float: left;
  color: #ddd;
}

.dEntry {
  max-width: 70%;
  background-color: rgb(20, 20, 30, 0.9);
  padding: 5%;
  margin: auto;
}



.diaryEntry {
  width: 700px;
  margin: auto;
  background-color: rgb(20, 20, 30, 0.9);
  padding: 40px;
  color: gray;
}

.invalid-feedback {
  color: #db3615;
}


.footer {
  color: rgb(247, 168, 23);
  background-color: #925925 !important;
  text-align: center;
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  text-align: center;
  padding-top: 10px;
  padding-bottom: 0px;
}

.notfound {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

body {
  background-image: url("./img/homepagebg.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  height: 890px;
}
