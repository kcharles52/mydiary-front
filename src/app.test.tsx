/// react libraries
import * as React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import  App  from './App';

describe("application",()=>{
    let shallowWrapper;
    beforeEach(()=>{
        shallowWrapper = shallow(<App />)
    })
    it("should render application",()=>{
        expect(shallowWrapper).toMatchSnapshot()
    })
})

