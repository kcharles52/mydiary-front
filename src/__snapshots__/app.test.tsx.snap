// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application should render application 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />,
        <AppRoutes />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoClose": 5000,
          "bodyClassName": null,
          "className": null,
          "closeButton": <CloseButton
            ariaLabel="close"
          />,
          "closeOnClick": true,
          "draggable": true,
          "draggablePercent": 80,
          "hideProgressBar": false,
          "newestOnTop": false,
          "pauseOnFocusLoss": true,
          "pauseOnHover": true,
          "position": "top-right",
          "progressClassName": null,
          "progressStyle": null,
          "rtl": false,
          "style": null,
          "toastClassName": null,
          "transition": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ToastContainer
            autoClose={5000}
            bodyClassName={null}
            className={null}
            closeButton={
              <CloseButton
                ariaLabel="close"
              />
            }
            closeOnClick={true}
            draggable={true}
            draggablePercent={80}
            hideProgressBar={false}
            newestOnTop={false}
            pauseOnFocusLoss={true}
            pauseOnHover={true}
            position="top-right"
            progressClassName={null}
            progressStyle={null}
            rtl={false}
            style={null}
            toastClassName={null}
            transition={[Function]}
          />,
          <AppRoutes />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoClose": 5000,
            "bodyClassName": null,
            "className": null,
            "closeButton": <CloseButton
              ariaLabel="close"
            />,
            "closeOnClick": true,
            "draggable": true,
            "draggablePercent": 80,
            "hideProgressBar": false,
            "newestOnTop": false,
            "pauseOnFocusLoss": true,
            "pauseOnHover": true,
            "position": "top-right",
            "progressClassName": null,
            "progressStyle": null,
            "rtl": false,
            "style": null,
            "toastClassName": null,
            "transition": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
